tests/integration/olp-cpp-sdk-dataservice-read/VersionedLayerClientTest.cpp

```diff
@@ -1866,4 +1902,691 @@ TEST_F(DataserviceReadVersionedLayerClientTest,
   ASSERT_TRUE(response.GetResult().empty());
 }
 
+TEST_F(DataserviceReadVersionedLayerClientTest, GetData404Error) {
+  olp::client::HRN hrn(GetTestCatalog());
+
+  EXPECT_CALL(
+      *network_mock_,
+      Send(IsGetRequest("https://blob-ireland.data.api.platform.here.com/"
+                        "blobstore/v1/catalogs/hereos-internal-test-v2/"
+                        "layers/testlayer/data/invalidDataHandle"),
+           _, _, _, _))
+      .WillOnce(ReturnHttpResponse(olp::http::NetworkResponse().WithStatus(404),
+                                   "Resource not found."));
+
+  auto catalog_client =
```

Comment:
```suggestion
  auto client =
```
Also in other places